### PR TITLE
Creating JobApiAction and JobApiActionImpl

### PR DESCRIPTION
**Pull request by Tauqueer031**

### CODE REVIEW

File: src/main/java/org/ideoholic/curium/model/employee/dto/ViewAllEmployeeResponseDto.java [hunk 1]
```diff
@@ -4,9 +4,12 @@
 
 import java.util.List;
 
+import org.ideoholic.curium.model.feescategory.dto.StudentListResponseDto;
```
Comment (reehanlalkhan) [inline src/main/java/org/ideoholic/curium/model/employee/dto/ViewAllEmployeeResponseDto.java]: No need of this import statement

File: src/main/java/org/ideoholic/curium/model/job/action/JobActionAdapter.java [hunk 2]
```diff
@@ -98,7 +99,7 @@ public void updateQueries() {
 	}
 
 	public void updateQueryRemarks() {
-		UpdateQueriesDto updateQueriesDto = new UpdateQueriesDto();
+		CombinedQueryDto updateQueriesDto = new CombinedQueryDto();
```
Comment (reehanlalkhan) [inline src/main/java/org/ideoholic/curium/model/job/action/JobActionAdapter.java]: All these need not be changed

File: src/main/java/org/ideoholic/curium/model/job/service/JobService.java [hunk 3]
```diff
@@ -603,11 +607,11 @@ public SearchStudentResponseDto toDoQueries(QueriesDto queriesDto,String userLog
         return searchStudentResponseDto;
 	}
 
-	public SearchStudentResponseDto updateQueryRemarks(UpdateQueriesDto updateQueriesDto,String userLoginId ) {
+	public SearchStudentResponseDto updateQueryRemarks(CombinedQueryDto combinedQueryDto,String userLoginId ) {
```
Comment (reehanlalkhan) [inline src/main/java/org/ideoholic/curium/model/job/service/JobService.java]: This change is not needed

File: src/main/java/org/ideoholic/curium/model/job/service/JobService.java [hunk 4]
```diff
@@ -616,13 +620,13 @@ public SearchStudentResponseDto updateQueryRemarks(UpdateQueriesDto updateQuerie
 		return searchStudentResponseDto;
 	}
 
-	public JobQueryDto viewTaskDetails(QueriesDto queriesDto,String branchId) {
+	public JobQueryDto viewTaskDetails(CombinedQueryDto combinedQueryDto,String branchId) {
```
Comment (reehanlalkhan) [inline src/main/java/org/ideoholic/curium/model/job/service/JobService.java]: This change is not needed

File: src/main/java/org/ideoholic/curium/model/job/dto/EmployeeResponseDto.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class EmployeeResponseDto {
```
Comment (reehanlalkhan) [inline src/main/java/org/ideoholic/curium/model/job/dto/EmployeeResponseDto.java:22]: This needs @JsonInclude(JsonInclude.Include.NON_ABSENT)

File: src/main/java/org/ideoholic/curium/model/job/dto/ViewStudentResponseDto.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
```
Comment (reehanlalkhan) [inline src/main/java/org/ideoholic/curium/model/job/dto/ViewStudentResponseDto.java]: Need to add @JsonInclude(JsonInclude.Include.NON_ABSENT)

File: src/main/java/org/ideoholic/curium/model/job/dto/ViewStudentResponseDto.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+public class ViewStudentResponseDto {
```
Comment (reehanlalkhan) [inline src/main/java/org/ideoholic/curium/model/job/dto/ViewStudentResponseDto.java]: Better name it TaskReportResponseDto

File: src/main/java/org/ideoholic/curium/model/job/action/JobApiActionImpl.java [hunk 8]
```diff
@@ -0,0 +1,426 @@
...
+
+	}
+
+	@PostMapping("/toDoTasks")
+	public ResponseEntity<JobQueryDto> toDoTasks(@RequestBody CombinedQueryDto queriesDto,
+			@RequestHeader(value = "userType") String userType, @RequestParam(value = "page") String page,
+			@RequestHeader(value = "branchid") String branchId, @RequestHeader(value = "username") String userName,
+			@RequestHeader(value = "userloginid") String userLoginId) {
+		jobService.toDoTasks(queriesDto, userLoginId);
+
```
Comment (reehanlalkhan) [inline src/main/java/org/ideoholic/curium/model/job/action/JobApiActionImpl.java:305]: At this point the data from CombinedQueryDto needs to be copied into individual objects of type that can be passed to jobService and viewTaskDetails

**Review by reehanlalkhan — CHANGES_REQUESTED** [review]

File: src/main/java/org/ideoholic/curium/model/employee/dto/ViewAllEmployeeResponseDto.java [hunk 9]
```diff
@@ -4,9 +4,12 @@
...
+
 @Data
 public class ViewAllEmployeeResponseDto {
```
Comment (reehanlalkhan) [inline src/main/java/org/ideoholic/curium/model/employee/dto/ViewAllEmployeeResponseDto.java]:
```suggestion
public class EmployeesWithSalaryResponseDto {
```

File: src/main/java/org/ideoholic/curium/model/job/action/JobActionAdapter.java [hunk 10]
```diff
@@ -84,6 +84,7 @@ public boolean viewAllQueriesDepartmentWise() {
 		request.setAttribute("queryList", jobQueryDto.getQueriesList());
 		request.setAttribute("noOfPages", jobQueryDto.getNoOfPages());
 		request.setAttribute("currentPage",jobQueryDto.getPage());
+		request.setAttribute("studentList", jobQueryDto.getStudentList());
```
Comment (reehanlalkhan) [inline src/main/java/org/ideoholic/curium/model/job/action/JobActionAdapter.java]: Remove this line

File: src/main/java/org/ideoholic/curium/model/job/action/JobActionAdapter.java [hunk 11]
```diff
@@ -175,6 +177,7 @@ public boolean viewAllTasksDepartmentWise() {
 		request.setAttribute("taskdetails", jobQueryDto.getTaskList());
 		request.setAttribute("noOfPages", jobQueryDto.getNoOfPages());
 		request.setAttribute("currentPage", jobQueryDto.getPage());
+		request.setAttribute("studentList", jobQueryDto.getTaskList());
```
Comment (reehanlalkhan) [inline src/main/java/org/ideoholic/curium/model/job/action/JobActionAdapter.java]: Remove this line

**Review by reehanlalkhan — CHANGES_REQUESTED** [review]

File: src/main/java/org/ideoholic/curium/model/job/action/JobActionAdapter.java (hunk 10 above)
Comment (Tauqueer031) [reply src/main/java/org/ideoholic/curium/model/job/action/JobActionAdapter.java]: done

File: src/main/java/org/ideoholic/curium/model/job/action/JobActionAdapter.java (hunk 11 above)
Comment (Tauqueer031) [reply src/main/java/org/ideoholic/curium/model/job/action/JobActionAdapter.java]: done

**Review by reehanlalkhan — DISMISSED** [review]

**Review by reehanlalkhan — APPROVED** [review]

**Pull request merged**